src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/ConsulLongPollCallback.java

```diff
@@ -185,21 +195,27 @@ private boolean contentChanged(byte[] newContent) {
         return !Arrays.equals(oldContent, newContent);
     }
 
-    private long updateIndex(Response response) {
+    private UpdateIndexStatus updateIndex(Response response) {
         String indexString = response.header("X-Consul-Index");
         if (indexString == null) {
             logger.error("There was no X-Consul-Index header in response for {} endpoint, retrying", endpoint);
-            return -1;
+            return UpdateIndexStatus.ERROR;
         }
 
         long index = Long.valueOf(indexString);
-        if (currentIndex.get() == index) {
+        long lastIndex = currentIndex.get();
+
+        if (index == lastIndex) {
             stats.indexNotChanged();
-            logger.trace("Discarding event on endpoint {} index {} as index did not change", endpoint, index);
-            return -1;
+            return UpdateIndexStatus.UNCHANGED;
+        } else if (index < lastIndex) {
+            stats.indexBackwards();
```

Comment:
I'd rather create a case in switch above for two unused enum values and handle "if bodies" there (like loggin etc), making this function only returning proper enum value, probably rename it to something like resolveIndexAction or smth. Currently 2 of enum values are beein ignored

Reply:
https://github.com/allegro/consul-recipes/pull/30/files - proposal

Reply:
I only meant for those 2 strings here

Reply:
Also I don’t feel that should be separate pr to extract two strings

Reply:
copied last 2 comments to the correct location:  https://github.com/allegro/consul-recipes/pull/29#discussion_r1039774930 - resolving this